pkg/lvm/lvm_util.go

```diff
@@ -265,6 +273,36 @@ func buildLVMDestroyArgs(vol *apis.LVMVolume) []string {
 	return LVMVolArg
 }
 
+// buildLVMRestoreThinSnapshotArgs returns lvcreate command for the restore volume from thin snapshot
+func buildLVMRestoreThinSnapshotArgs(vol *apis.LVMVolume) []string {
+	var LVMRestoreThinVolArg []string
+
+	// command to create restored thin volume from thin snapshot
+	// `lvcreate -s -n <restore-volume-name>  lvmvg/<thin-snapshot-name> -y`
+	LVMRestoreThinVolArg = append(LVMRestoreThinVolArg, "-s", "-n", vol.Name)
+
+	// update args with <volgroup>/<thin-snapshot-name>
+	LVMRestoreThinVolArg = append(LVMRestoreThinVolArg, vol.Spec.VolGroup+"/"+getLVMSnapName(vol.Spec.Source))
+
+	// -y to automatically answer "yes" to all prompts
+	LVMRestoreThinVolArg = append(LVMRestoreThinVolArg, "-y")
```

Comment:
we could add a comment here about wipesignatures being skipped, just for completeness